libmachine/mcnutils/b2d.go

```diff
@@ -304,10 +306,19 @@ func (b *b2dISO) version() (string, error) {
 		return "", err
 	}
 
-	verRegex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
-	ver := string(verRegex.Find(isoMetadata))
-	log.Debug("local Boot2Docker ISO version: ", ver)
-	return ver, nil
+	fullVersion := string(isoMetadata)
+
+	versionIndex := strings.Index(fullVersion, versionPrefix)
+	if versionIndex == -1 {
+		return "", fmt.Errorf("Did not find prefix %q in version string", versionPrefix)
+	}
+
+	// Original magic file string looks similar to this: "Boot2Docker-v0.1.0              "
+	// This will return "v0.1.0" given the above string
+	vers := strings.TrimSpace(fullVersion)[versionIndex+1:]
```

Comment:
If you come back to this code, it might be a good idea to move the `[]` inside the `()` here to make sure we don't get messed up by leading whitespace in the future. :+1:

(ie, `fullVersion = " something-v1.0.0 "` would lead to `1.0.0` instead of `v1.0.0`)